google-cloud-spanner/lib/google/cloud/spanner/service.rb

```diff
@@ -37,10 +37,11 @@ class Service
 
         ##
         # Creates a new Service instance.
-        def initialize project, credentials,
+        def initialize project, credentials, quota_project: nil,
                        host: nil, timeout: nil, lib_name: nil, lib_version: nil
           @project = project
           @credentials = credentials
+          @quota_project_id = quota_project || credentials.quota_project_id if credentials.respond_to? :quota_project_id
```

Comment:
This will ignore the passed in `quota_project` if `credentials` does not respond to `:quota_project_id`.